infrastructure/lib/stacks/gitHubAutomationApp.ts

```diff
@@ -0,0 +1,138 @@
+
+
+        const launchConfiguration = this.asg.node.findChild('LaunchConfig') as CfnLaunchConfiguration;
+        launchConfiguration.metadataOptions = {
+            httpPutResponseHopLimit: 2,
+            httpEndpoint: "enabled",
+            httpTokens: "required"
+        };
+
+        const instanceName = 'OpenSearchMetrics-GitHubAutomationApp-Host';
+        Aspects.of(this.asg).add(new Tag('name', instanceName, {
+            applyToLaunchedInstances: true,
+            includeResourceTypes: ['AWS::AutoScaling::AutoScalingGroup']
+        }),);
+        this.asg.addUserData(...this.getUserData(props.secret.secretName));
+    }
+
+    private createInstanceRole(secretArn: string, account: string): Role {
+        const role = new Role(this, "OpenSearchMetrics-GitHubAutomationApp-Role", {
+            assumedBy: new CompositePrincipal(
+                new ServicePrincipal('ec2.amazonaws.com'),
+                new ArnPrincipal(`arn:aws:iam::${account}:role/OpenSearchMetrics-GitHubAutomationApp-Role`)
+            ),
+            roleName: "OpenSearchMetrics-GitHubAutomationApp-Role",
+        });
+        role.addManagedPolicy(ManagedPolicy.fromAwsManagedPolicyName('AmazonSSMManagedInstanceCore'));
+        role.addToPolicy(
+            new PolicyStatement({
+                effect: Effect.ALLOW,
+                actions: ["secretsmanager:GetSecretValue"],
+                resources: [secretArn],
+            }),
+        );
+        role.addToPolicy(
+            new PolicyStatement({
+                effect: Effect.ALLOW,
+                actions: ["sts:AssumeRole"],
+                resources: [role.roleArn],
+            }),
+        );
+        return role;
+    }
+
+
+    private getUserData(secretName: string): string[] {
+        return [
+            'sudo yum install -y https://s3.amazonaws.com/ec2-downloads-windows/SSMAgent/latest/linux_amd64/amazon-ssm-agent.rpm',
+            'sudo dnf update -y',
+            'sudo yum install git docker -y',
+            'sudo systemctl enable docker',
+            'sudo systemctl start docker',
+            'sudo curl -L https://github.com/docker/compose/releases/latest/download/docker-compose-$(uname -s)-$(uname -m) -o /usr/local/sbin/docker-compose',
+            'sudo chmod a+x /usr/local/sbin/docker-compose',
+            'git clone https://github.com/opensearch-project/automation-app.git',
+            `aws secretsmanager get-secret-value --secret-id ${secretName} --query SecretString --output text >> automation-app/.env`,
+            'cd automation-app/docker',
+            'PORT=8080 RESOURCE_CONFIG=configs/resources/opensearch-project-resource.yml OPERATION_CONFIG=configs/operations/github-merged-pulls-monitor.yml docker-compose -p automation-app-1 up -d',
+            'PORT=8081 RESOURCE_CONFIG=configs/resources/opensearch-project-resource.yml OPERATION_CONFIG=configs/operations/github-workflow-runs-monitor.yml docker-compose -p automation-app-2 up -d'
```

Comment:
Nit: maybe update the name to be in sync with your operation name?

Reply:
Updated @peterzhuamazon.